src/sctools/test/test_groups.py

```diff
@@ -26,232 +26,272 @@ def check_parsed_metrics_csv(file_name, cell_id, class_name, expected_metrics):
             ]
         ),
         OrderedDict(
             [
-                ('Sample', 'test'),
-                ('ALT_BASE', 'T'),
-                ('ALT_COUNT', '16'),
-                ('REF_BASE', 'A'),
-                ('REF_COUNT', '231512'),
-                ('SUBSTITUTION', 'A>T'),
-                ('SUBSTITUTION_RATE', '6.9e-05'),
+                ("Sample", "test"),
+                ("ALT_BASE", "T"),
+                ("ALT_COUNT", "16"),
+                ("REF_BASE", "A"),
+                ("REF_COUNT", "231512"),
+                ("SUBSTITUTION", "A>T"),
+                ("SUBSTITUTION_RATE", "6.9e-05"),
             ]
         ),
         OrderedDict(
             [
-                ('Sample', 'test'),
-                ('ALT_BASE', 'A'),
-                ('ALT_COUNT', '16'),
-                ('REF_BASE', 'C'),
-                ('REF_COUNT', '173880'),
-                ('SUBSTITUTION', 'C>A'),
-                ('SUBSTITUTION_RATE', '9.2e-05'),
+                ("Sample", "test"),
+                ("ALT_BASE", "A"),
+                ("ALT_COUNT", "16"),
+                ("REF_BASE", "C"),
+                ("REF_COUNT", "173880"),
+                ("SUBSTITUTION", "C>A"),
+                ("SUBSTITUTION_RATE", "9.2e-05"),
             ]
         ),
         OrderedDict(
             [
-                ('Sample', 'test'),
-                ('ALT_BASE', 'G'),
-                ('ALT_COUNT', '14'),
-                ('REF_BASE', 'C'),
-                ('REF_COUNT', '173880'),
-                ('SUBSTITUTION', 'C>G'),
-                ('SUBSTITUTION_RATE', '8.1e-05'),
+                ("Sample", "test"),
+                ("ALT_BASE", "G"),
+                ("ALT_COUNT", "14"),
+                ("REF_BASE", "C"),
+                ("REF_COUNT", "173880"),
+                ("SUBSTITUTION", "C>G"),
+                ("SUBSTITUTION_RATE", "8.1e-05"),
             ]
         ),
         OrderedDict(
             [
-                ('Sample', 'test'),
-                ('ALT_BASE', 'T'),
-                ('ALT_COUNT', '82'),
-                ('REF_BASE', 'C'),
-                ('REF_COUNT', '173880'),
-                ('SUBSTITUTION', 'C>T'),
-                ('SUBSTITUTION_RATE', '0.000471'),
+                ("Sample", "test"),
+                ("ALT_BASE", "T"),
+                ("ALT_COUNT", "82"),
+                ("REF_BASE", "C"),
+                ("REF_COUNT", "173880"),
+                ("SUBSTITUTION", "C>T"),
+                ("SUBSTITUTION_RATE", "0.000471"),
             ]
         ),
     ]
 
-    with open('output_picard_group_error_summary_metrics.csv') as f:
+    with open("output_picard_group_error_summary_metrics.csv") as f:
         reader = csv.DictReader(f)
+
+        i = 0
+        match_list = []
         for line in reader:
-            assert line in expected_metrics
-    os.remove('output_picard_group_error_summary_metrics.csv')
+            for ordered_dict in expected_metrics:
```

Comment:
We should make it clear that this test no longer considers the order of the Ordered Dictionary objects

Reply:
If we are ok with ignoring the order of the metrics in the file, you can check the ordered dictionary from the file against a non-ordered dictionary from expected using 'in' and the order will not be considered. We do not need the extra method to compare ordered dictionaries

Reply:
changed to unordered